docs/contributing.md

```diff
@@ -0,0 +1,117 @@
+
+Once the pull request is ready, the author can switch from draft to regular pull request.
+
+At this point, the pull request will be triaged during the next triage session and reviewers will be assigned to it.
+
+### Merging
+
+Every PR requires at least one review to be merged.
+
+If the author has the editing rights to the repository merging should be performed by the author of the pull request. If the author wants to grant another team member rights to merge, they can state so in the PR comment.
+
+# Review Model
+
+Every project has its own code authoring and review culture which grows organically as the project matures and is a subject to change.
+Below is the description of the model we try to follow, with exceptions when the common sense dictates otherwise.
+
+## Review Types
+
+Review can be of either **architectural** or **technical** value, and often is of both.
+
+The PR author can specify, when requesting review, what kind of review they are asking for, from each reviewer, or even which area they'd like the reviewer to focus on (useful when distributing reviews).
+
+## Reviewer Role
+
+The reviewer is responsible for accepting a pull request only once they feel the current PR is ready to be merged even if their comments were not to be applied.
+
+The *accepted* can be set with pending review comments, if those comments don't affect whether the patch is ready to be merged (for example, they're stylistic suggestions).
+
+The should communicate the nature of their review comments - specifically, between the three types: *"blocking"*, *"suggestion"*, and *"optional"*.
+* **blocking** is when the reviewer considers the change to be unmergable and requires a new revision.
+* **suggestion** is for when the reviewer considers the change to be suboptimal, but usable, and wants to defer the decision to the PR author, while stating their opinion.
+* **optional** is for when the reviewer considers multiple options to be mostly comparable or tradeoffs, and wants to defer to the PR author for the final decision after bringing up a new option.
+
+## Social Contract
+
+`ICU4X` project focuses on a fairly hermetic domain of software internationalization, which requires prior knowledge of the domain.
+With that in mind, most engineers working on patch authoring and reviews are expected to be senior enough to be trusted with the quality of their additions to the code.
+
+For those reasons, we are primarily placing **trust** in pull request authors to write high quality, readable, tested, maintainable and well documented code.
+
+The role of the **reviewer** in such model is more conservative and is reduced to verification of the code from a particular angle with minimal impact on the pull request.
+Examples of such angle may be:
+
+* How the PR fits into the component public API
+* Memory management of the code in pull request
+* Test coverage, and sanity checks
+* Use of CLDR, Unicode, Rust and other best practices
```

Comment:
Also, consistency with the ICU4X style guide, and other conventions like locale data management, etc.